RTG acceleration model

jo11he opened the pull request:
This PR integrates the RTG Acceleration model feature into develop branch. 
Unit tests are now complete and running.

jo11he committed:
[untested] add rtg acceleration type, settings and model

jo11he committed:
add rtg_acceleration required environment updates

jo11he committed:
update include in headers

jo11he committed:
add environment dependencies of rtg_acceleration acceleration type

jo11he committed:
implement ModelSetup unit test for rtg acceleration

jo11he committed:
add successful implementation of setup and numerical unit tests, with tested verson of acceleration model class

jo11he committed:
add rtg_force_vector to EstimatableParameter enum

jo11he committed:
add rtgForceVector EstimatableParameter class and implement necessary getters in RTGAccelerationModel class

jo11he committed:
Add constructor for EstimatableParameterSettings class that instantiates with rtg_force_vector type

jo11he committed:
include switch case for creation of rtgForceVector parameter into the createEstimatableParameters.h file

jo11he committed:
add rtg force magnitude as separate parameter

jo11he committed:
establish RTGAccelerationPartials class with partials for rtg force vector and magnitude parameters

jo11he committed:
establish and expose attributes for partials computation

jo11he committed:
include new partial in cmake lists

jo11he committed:
resolve casting conflicts, migrate magnitude parameter to double section

jo11he committed:
add factory function for rtg_acceleration type settings

jo11he committed:
add factory functions for rtg paraeter settings

jo11he committed:
remove redundant factory function

jo11he committed:
cut integration time of acceleration unit test

jo11he committed:
add working iso unit test and update rtg acceleration partials

jo11he committed:
add rtg acceleration model, partials and working estimation unit test

jo11he committed:
extend scope of unit test

jo11he committed:
Merge branch 'feature/rtg_acceleration' into develop

jo11he committed:
eliminate double mention of parameter type

jo11he committed:
adjust order of attribute/member declaration to match order of initialisation

jo11he committed:
extend scope of RTG Estimation Unit Test

jo11he committed:
correct parameter description

jo11he committed:
clean up cout from troubleshooting

jo11he committed:
add user controlled unit conversion on panel model from file

DominicDirkx commented:
@jo11he Looks like the Windows build is failing due to the use of M_PI. This is a platform-specific macro not available on Windows. If you switch to the use of mathematical_constants::PI, it should work fine. Since the Mac and Linux CI pass without problems, I suspect this will be a quick fix :) The same issue occurs on https://github.com/tudat-team/tudatpy/pull/378

jo11he committed:
resolve reference issue in RTG Acceleration Model

jo11he committed:
replace M_PI with universal mathematical_constants::PI

jo11he committed:
add default input unit to bodyPanelSettingsListFromDae for backwards compatibility

jo11he commented:
This includes the fix of the dangling rtg force reference. RTG force modeling and estimation is also validated on Cassini.
I addressed the M_PI issues in the unit tests and added the default value to the paneled model input_unit argument.

If CI and windows tests pass this time, this is ready to merge, effectively including:
- validated RTG Force Model + Parameter
- paneled model input unit argument
